src/battle.ts

```diff
@@ -648,8 +648,13 @@ class Side {
 			if (this.foe && this.avatar === this.foe.avatar) this.rollTrainerSprites();
 		}
 	}
-	addSideCondition(effect: Effect) {
+	addSideCondition(effect: Effect, info?: [string, number, number, number]) {
 		let condition = effect.id;
+		if (info) {
+			this.sideConditions[condition] = info;
+			this.battle.scene.addSideCondition(this.n, condition);
+			return;
+		}
```

Comment:
Considering this shares zero code with the other `addSideCondition` code, you should probably implement this as a separate `switchSides` function.